modules/fluid_properties/unit/include/CaloricImperfectGasTest.h

```diff
@@ -0,0 +1,176 @@
+    _fp_bad_h_fn2 = &_fe_problem->getUserObject<CaloricImperfectGas>("fp_bad_h_fn2");
+
+    // compare ideal gas and calorically imperfect gas
+    InputParameters ideal_uo_pars = _factory.getValidParams("IdealGasFluidProperties");
+    ideal_uo_pars.set<Real>("molar_mass") = 0.002;
+    ideal_uo_pars.set<Real>("gamma") = 1.41;
+    _fe_problem->addUserObject("IdealGasFluidProperties", "ideal_fp", ideal_uo_pars);
+    _ideal = &_fe_problem->getUserObject<IdealGasFluidProperties>("ideal_fp");
+
+    Real Ru = 8.31446261815324;
+    Real Rs = Ru / 0.002;
+    Real gamma = 1.41;
+    Real cp_val = gamma / (gamma - 1) * Rs;
+    Real cv_val = cp_val / gamma;
+    T = {0.0, 500.0};
+    h = {0.0, 500.0 * cp_val};
+    e = {0.0, 500.0 * cv_val};
+    cp = {cp_val, cp_val};
+    cv = {cv_val, cv_val};
+    InputParameters lin_h_fn_params = _factory.getValidParams("PiecewiseLinear");
+    lin_h_fn_params.set<std::vector<Real>>("x") = T;
+    lin_h_fn_params.set<std::vector<Real>>("y") = h;
+    _fe_problem->addFunction("PiecewiseLinear", "lin_h_fn", lin_h_fn_params);
+
+    InputParameters lin_e_fn_params = _factory.getValidParams("PiecewiseLinear");
+    lin_e_fn_params.set<std::vector<Real>>("x") = T;
+    lin_e_fn_params.set<std::vector<Real>>("y") = e;
+    _fe_problem->addFunction("PiecewiseLinear", "lin_e_fn", lin_e_fn_params);
+
+    InputParameters const_cp_fn_params = _factory.getValidParams("PiecewiseLinear");
+    const_cp_fn_params.set<std::vector<Real>>("x") = T;
+    const_cp_fn_params.set<std::vector<Real>>("y") = cp;
+    _fe_problem->addFunction("PiecewiseLinear", "const_cp_fn", const_cp_fn_params);
+
+    InputParameters const_cv_fn_params = _factory.getValidParams("PiecewiseLinear");
+    const_cv_fn_params.set<std::vector<Real>>("x") = T;
+    const_cv_fn_params.set<std::vector<Real>>("y") = cv;
+    _fe_problem->addFunction("PiecewiseLinear", "const_cv_fn", const_cv_fn_params);
+
+    InputParameters compare_uo_pars = _factory.getValidParams("CaloricImperfectGas");
+    compare_uo_pars.set<Real>("molar_mass") = 0.002;
+    compare_uo_pars.set<FunctionName>("h") = "lin_h_fn";
+    compare_uo_pars.set<FunctionName>("e") = "lin_e_fn";
+    compare_uo_pars.set<FunctionName>("cp") = "const_cp_fn";
+    compare_uo_pars.set<FunctionName>("cv") = "const_cv_fn";
+    compare_uo_pars.set<FunctionName>("mu") = "default_fn";
+    compare_uo_pars.set<FunctionName>("k") = "default_fn";
+    compare_uo_pars.set<Real>("min_temperature") = 100.0;
+    compare_uo_pars.set<Real>("max_temperature") = 500.0;
+    _fe_problem->addUserObject("CaloricImperfectGas", "compare_fp", compare_uo_pars);
+    _compare_with_ideal = &_fe_problem->getUserObject<CaloricImperfectGas>("compare_fp");
+  }
+
+  const CaloricImperfectGas * _fp;
+  const CaloricImperfectGas * _fp_bad_h_fn;
+  const CaloricImperfectGas * _fp_bad_h_fn2;
+  const IdealGasFluidProperties * _ideal;
+  const CaloricImperfectGas * _compare_with_ideal;
```

Comment:
Did you try to put each group of tests into their own fixture (i.e. `TEST_F`). You would construct the `CaloricImperfectGas` class, set it up, and ran your tests. That would be the optimal case.

Was there something you were not able to do, so you dropped the first 3 into one fixture and the ideal gas comparison into another one?

If it is too late to build the UOs and Functions in the fixture, at least split the `testAll` into 3.